core/remotes/docker/fetcher.go

```diff
@@ -262,7 +265,27 @@ func (r dockerFetcher) FetchByDigest(ctx context.Context, dgst digest.Digest, op
 	return seeker, desc, nil
 }
 
+const (
+	_        = iota
+	kB int64 = 1 << (10 * iota)
+	mB
+	gB
+	tB
+	pB
+	eB
+
+	// When under this size, a layer will not be fetched using parallel requets
+	minimumSizeForParallelDL = 1 * mB
+)
+
 func (r dockerFetcher) open(ctx context.Context, req *request, mediatype string, offset int64) (_ io.ReadCloser, retErr error) {
+	parallelism, chunkSize := r.config.Parallelism(), int64(r.config.ConcurrentFetchChunksSizeMB)*mB
+	if r.config.Limiter != nil {
+		if err := r.config.Limiter.Acquire(ctx, parallelism); err != nil {
+			return nil, err
+		}
+	}
```

Comment:
Nit: How about having a defer block to do `r.config.Limiter.Release(parallelism)` here?